pandas/core/arrays/categorical.py

```diff
@@ -1081,20 +1081,26 @@ def remove_unused_categories(self, inplace=False):
             return cat
 
     def map(self, mapper):
-        """Apply mapper function to its categories (not codes).
+        """
+        Map categories (not codes) using input correspondence (a dict,
+        Series, or function).
+
+        Maps the categories to new categories. If the mapping
+        correspondence maps each original category to a different new category
+        the result is a Categorical which has the same order property as
+        the original, otherwise an np.ndarray is returned.
+
+        If a dictionary or Series is used any unmapped category is mapped to
+        NA. Note that if this happens an np.ndarray will be returned.
 
         Parameters
         ----------
         mapper : callable
-            Function to be applied. When all categories are mapped
-            to different categories, the result will be Categorical which has
-            the same order property as the original. Otherwise, the result will
-            be np.ndarray.
+            Function to be applied.
 
         Returns
         -------
         applied : Categorical or Index.
```

Comment:
No need for a period at the end of the type. Also should namespace pandas types (except for DataFrame and Series) so `pandas.Categorial or pandas.Index`

Reply:
This is outside the original scope of the contribution, see first PR comment.